### DISCUSSION TITLE
Possible to visit Modal page regularaly?

### DISCUSSION THREAD

**Discussion by AUAboi** [inertiaui/modal#68] [Q&A] [answered]
I want to be able to allow my users to visit the page wrapped in Modal component by just visiting the URL. Right now it throws error if Modal page is visited manually. Is there anyway to accomplish both normal behavior and modal in same template?

**Comment by pascalbaljet** [comment, marked as answer]
Try using a [dynamic component](https://vuejs.org/guide/essentials/component-basics#dynamic-components) to determine whether the page should be wrapped in a `Modal` component or not. Something like this:

```vue
<script setup>
import { Modal } from '@inertiaui/modal-vue'
import { ref } from 'vue';

const isModal = ref(true);
</script>

<template>
    <component :is="isModal ? Modal : 'div'">
        <!-- content -->
    </component>
</template>
```

I'm converting this issue into a discussion as it's not really a bug.
